File: qtcopy_files/initapp.py
import os

DATA_DIR = "data"
TARGET_DIR_PATH = f"{DATA_DIR}/targetdir.txt"

def init_app():
    if not os.path.exists(DATA_DIR):
        os.makedirs(DATA_DIR)

    if not os.path.exists(TARGET_DIR_PATH):
        with open(TARGET_DIR_PATH, 'w') as f:
            f.write("select/target/dir")

__all__ = [TARGET_DIR_PATH, init_app]

File: src/initapp.cpp
#include "initapp.h"
#include <filesystem>
#include <fstream>

void init_app() {
  if (!std::filesystem::exists(DATA_DIR)) {
    std::filesystem::create_directories(DATA_DIR);
  }

  if (!std::filesystem::exists(TARGET_DIR_PATH)) {
    std::ofstream outfile(TARGET_DIR_PATH);
    outfile << "select/target/dir";
    outfile.close();
  }
}
